src/Features/Core/Portable/CodeRefactoringHelpers.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static class CodeRefactoringHelpers
+    {
+        public static Task<bool> RefactoringSelectionIsValidAsync(
+            Document document, TextSpan selection, SyntaxNode node, CancellationToken cancellation)
+        {
+            return RefactoringSelectionIsValidAsync(document, selection, node, ImmutableArray<SyntaxNode>.Empty, cancellation);
+        }
+
+        public static async Task<bool> RefactoringSelectionIsValidAsync(
+            Document document, TextSpan selection, SyntaxNode node,
+            ImmutableArray<SyntaxNode> holes, CancellationToken cancellationToken)
```

Comment:
it's worth having a comment here explaining the overall concept as well as how things like 'holes' work.